iceoryx_examples/icediscovery_in_c/sleep_for.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_BINDING_C_SLEEP_FOR_H
+#define IOX_BINDING_C_SLEEP_FOR_H
+
+#ifdef _WIN32
+#include <windows.h>
+
+void sleep_for(uint32_t milliseconds)
+{
+    Sleep((uint64_t)milliseconds);
+}
+
+#else
+#include <unistd.h>
+
+void sleep_for(uint32_t milliseconds)
```

Comment:
Is the choice of `uint32_t` here deliberate to be able to safely multiply with 1000 and get a non-overflowing value? I think we can have an overflow in the multiplication and then things get weird (of course, this requires fairly large values > 2^32 /10³) 

According to https://man7.org/linux/man-pages/man3/usleep.3.html `useconds_t` appears to be an `unsigned int` (usually). But portability is shaky here anyway, we cannot do much except to work with microseconds in the call.

But it is only example code and not called with large values. If it were library code we would need to make it safer.

TLDR: For a lib we would need to make it safer by using `uint32_t` microseconds (and the windows call would need to perform a division by 1000). Not needed in example here.

Reply:
@FerdinandSpitzschnueffler I think this is of no concern as we use it in the example and can be resolved.